client/my-sites/site-settings/site-owner-transfer/site-owner-user-search.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, Gridicon } from '@automattic/components';
+import styled from '@emotion/styled';
+import { TextControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useState, FormEvent, useEffect } from 'react';
+import { useCheckSiteTransferEligibility } from './use-check-site-transfer-eligibility';
+
+const TextControlContainer = styled.div( {
+	marginBottom: '2em',
+} );
+
+const Error = styled.div( {
+	display: 'flex',
+	alignItems: 'center',
+	color: 'red',
+} );
+
+const ErrorText = styled.p( {
+	margin: 0,
+	marginLeft: '0.4em',
+	fontSize: '14px',
+} );
+
+const SiteOwnerTransferEligibility = ( {
+	selectedSiteId,
+	selectedSiteSlug,
+	siteOwner,
+	onNewUserOwnerSubmit,
+}: {
+	selectedSiteId: number;
+	selectedSiteSlug: string;
+	siteOwner: string;
+	onNewUserOwnerSubmit: ( user: string ) => void;
+} ) => {
+	const translate = useTranslate();
+	const [ tempSiteOwner, setTempSiteOwner ] = useState( siteOwner );
+	const [ siteTransferEligibilityError, setSiteTransferEligibilityError ] = useState( '' );
+
+	const { checkSiteTransferEligibility, isLoading: isCheckingSiteTransferEligibility } =
+		useCheckSiteTransferEligibility( selectedSiteId, {
+			onMutate: () => {
+				setSiteTransferEligibilityError( '' );
+			},
+			onError: ( e ) => {
+				setSiteTransferEligibilityError( e.message );
+			},
+			onSuccess: () => {
+				onNewUserOwnerSubmit?.( tempSiteOwner || '' );
+			},
+		} );
+
+	useEffect( () => {
+		setTempSiteOwner( siteOwner );
+	}, [ siteOwner ] );
+
+	const handleFormSubmit = ( event: FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+		if ( ! tempSiteOwner ) {
+			return;
+		}
+		checkSiteTransferEligibility( { newSiteOwner: tempSiteOwner } );
+	};
+
+	return (
+		<form onSubmit={ handleFormSubmit }>
+			<p>
+				{ translate(
+					'Please enter the username or email address of the registered WordPress.com user that you want to transfer ownership of %(siteSlug)s to:',
+					{
+						args: { siteSlug: selectedSiteSlug },
```

Comment:
Let's make the domain name bold so that it stands out from the text like so:

<img width="677" alt="Screenshot 2023-05-31 at 10 11 53 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/0e22aa8f-2ac8-4ec4-9cab-b02c021ab4b3">

This is what I am seeing at the moment:

<img width="725" alt="Screenshot 2023-05-31 at 10 12 30 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/299b6038-e951-4c99-9344-f5ae9f0b131d">